### PR TITLE
Allow extending any, without noImplicitAny errors

### LINKED ISSUE #22443: Javascript should allow you to extend any
```ts
// @noEmit: true
// @allowJs: true
// @checkJs: true
// @Filename: defaults.js
/** @type {any} */
var X
class Y extends X {
    constructor() {
        super()
    }
}
```

**Expected behavior:**
No error, or at least an error on X "Can't extend value of type any".

**Actual behavior:**
Error on `super()`, "Call target does not contain any signatures."

There are lots of anys floating around Javascript, so this should be allowed, even if it's not trivial. Maybe it should be allowed in Typescript too.

### PR DESCRIPTION
**Pull request by sandersn**
Whenever there is a reference to super, or a reference to a property on `this` that is unknown, you will get an implicit any error. Previously you would only an error on calling `super()`, but it always appeared, not just with `--noImplicitAny`.

After discussing it with @weswigham and @rbuckton, I decided that an assertion escape hatch for super was too problematic to add.

Edit: After further discussion, we decided to always allow extending any.

Fixes #22443

### CODE REVIEW

File: src/compiler/checker.ts [hunk 1]
```diff
@@ -14288,9 +14298,13 @@ namespace ts {
                 return unknownType;
             }
 
-            return nodeCheckFlag === NodeCheckFlags.SuperStatic
+            const type = nodeCheckFlag === NodeCheckFlags.SuperStatic
                 ? getBaseConstructorTypeOfClass(classType)
                 : getTypeWithThisArgument(baseClassType, classType.thisType);
+            if (noImplicitAny && type.flags & TypeFlags.Any) {
```
[the remark this answers is not in the document]
Comment (mhegazy) [reply src/compiler/checker.ts]: then we do not need an error altogether..

[the remark this answers is not in the document]
Comment (mhegazy) [reply src/compiler/checker.ts]: Our methodology here was to issue one error per declaration, and not per use.. that is why you only get one error on:
```ts
var a;  // error
var b = a();  // No error here
var c = a.b.c; // no error here
```

[the remark this answers is not in the document]
Comment (mhegazy) [reply src/compiler/checker.ts]: in this PR we are making every reference to `super`/ `this` an error.. do not think ppl need that much nagging.. they get it, it is `any`.

[the remark this answers is not in the document]
Comment (sandersn) [reply src/compiler/checker.ts]: After some discussion we decided to drop the error. In normal javascript usage, as soon as you turn on `--noImplicitAny` you will get an error on untyped modules, after which you will have to assume that anything from that module is also of type any.

**Pull request merged**